src/instrumentations/external/nocode/nocode.ts

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as path from 'path';
+import * as fs from 'fs';
+import {
+  InstrumentationBase,
+  InstrumentationModuleDefinition,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+import { VERSION } from '../../../version';
+import { context, trace } from '@opentelemetry/api';
+
+export interface AttributeDefinition {
+  attrIndex: number;
+  attrPath: string;
+  key: string;
+}
+
+export interface InstrumentationFileDefinition {
+  name: string;
+  method: string;
+  spanName?: string;
+  attributes?: AttributeDefinition[];
+}
+
+export interface InstrumentationDefinition {
+  moduleName?: string;
+  absolutePath?: string;
+  supportedVersions?: string[];
+  files: InstrumentationFileDefinition[];
+}
+
+export interface NoCodeInstrumentationConfig extends InstrumentationConfig {
+  definitions?: InstrumentationDefinition[];
+}
+
+export class NoCodeInstrumentation extends InstrumentationBase<NoCodeInstrumentationConfig> {
+  constructor(config: NoCodeInstrumentationConfig = {}) {
+    super('splunk-opentelemetry-instrumentation-nocode', VERSION, config);
+  }
+
+  protected init(): InstrumentationModuleDefinition[] {
+    const configPath = process.env.NOCODE_CONFIG_PATH
+      ? path.normalize(process.env.NOCODE_CONFIG_PATH)
+      : path.resolve(process.cwd(), 'nocode.config.json');
+
+    this._diag?.debug?.(
+      `Loading NoCodeInstrumentation config from ${configPath}`
+    );
+
+    try {
+      const fileContent = fs.readFileSync(configPath, 'utf-8');
```

Comment:
no check if file exists before trying to read it -> always throws an error if `./nocode.config.json` doesn't exist (pretty much the default case) -> gets output as an error in console

https://github.com/signalfx/splunk-otel-js/actions/runs/16619852205/job/47021369811?pr=1033#step:7:543